packages/legend-query-builder/src/stores/QueryBuilderValueSpecificationBuilderHelper.ts

```diff
@@ -150,5 +185,12 @@ export type LambdaFunctionBuilderOption = {
    * queryBuilderState will make the lambda function building process overrides several query values, such as the row limit.
    */
   isBuildingExecutionQuery?: boolean | undefined;
+
+  /**
+   * Set this to `true` when we construct query for execution within the app to view preview data.
+   * queryBuilderState will make the lambda function building process overrides `.all()` to `.allVersions()` if the Class
+   * is milestoned and `property` to `propertyAllVersions` if a property is milestoned.
+   */
+  isBuildingExecutionQueryToPreviewData?: boolean | undefined;
```

Comment:
is it possible that we give it a more generic name? We also need it for checking entitlements right?

Reply:
@YannanGao-gs any suggestions for the name. 
should we make it specific to milestonining? maybe like
`useAllVersionsForMilestoning` ?